packages/server/src/sdk/app/rows/internal.ts

```diff
@@ -20,21 +20,25 @@ export async function save(
   const { tableId, viewId } = tryExtractingTableAndViewId(tableOrViewId)
   inputs.tableId = tableId
 
+  let source: Table | ViewV2
+  let table: Table
+  if (viewId) {
+    source = await sdk.views.get(viewId)
+    table = await sdk.views.getTable(viewId)
+  } else {
+    source = await sdk.tables.getTable(tableId)
+    table = source
+  }
+
   if (!inputs._rev && !inputs._id) {
     inputs._id = db.generateRowID(inputs.tableId)
   }
 
-  // this returns the table and row incase they have been updated
-  const dbTable = await sdk.tables.getTable(inputs.tableId)
-
-  // need to copy the table so it can be differenced on way out
-  const tableClone = cloneDeep(dbTable)
```

Comment:
Unsure what this cloning was about - I get the feeling the `inputProcessing` can make changes to the schema etc in someway of the passed in table/source, has this changed and we don't need to worry about it anymore?

Realistically we shouldn't need to worry about this, but just raising it as this likely was added to fix a bug at some point.

Reply:
You're correct, yeah. I address this in another comment, but `inputProcessing` can modify the table in very rare cases and we leak that out through to the caller and leave them to handle it. This PR changes that so that `inputProcessing` saves the table itself now.